### PR TITLE
  #184581188 A User Should be able to list all Items 

### PR DESCRIPTION
**Pull request by wilbrord2**
#### What does this PR do?
This PR help A user/Seller to get a list of product in his/her collection

#### Description of Task to be completed?
- Creating route to get items in seller collection
- protecting route
- testing route
- document route

#### How should this be manually tested?
- Clone this repo git clone https://github.com/atlp-rwanda/e-comm-team-amigos-bn
- Switch your working directory cd e-comm-team-amigos-bn
- Create an .env file and update it with .env-example
- Install all the dependencies by running npm install
- make sure you have testing tool like `Postman`
- Then run the project by typing `npm run dev`
- you should create a user with a role of `vendor`

#### Endpoints to be tested:
- `/user/login/`
- `/user/otp/`
- `/product/create/`
- `/product/collection/`

#### Any background context you want to provide?
- Have Node Js installed in your local machine

#### What are the relevant pivotal tracker/Trello stories?

Story ID: [#184581188](https://www.pivotaltracker.com/story/show/184581188)

#### All Test are passing
Screenshot (142)

#### Endpoint in documentation
Screenshot (141)

### CODE REVIEW

File: src/controllers/product.controller.js [hunk 1]
```diff
@@ -29,30 +28,32 @@ export const getAllProduct = async (req, res) => {
 export const getAvailableProducts = async (req, res) => {
   try {
     const products = await models.Product.findAll({
-      where: { available: true }
+      where: { available: true },
     });
-    res.status(200).json({response:products});
+    res.status(200).json({ response: products });
   } catch (err) {
     console.error(err);
```
Comment (Emile-SHUMBUSHO) [inline src/controllers/product.controller.js]: not good to push console.log statement in the version control.

File: src/controllers/product.controller.js [hunk 2]
```diff
@@ -29,30 +28,32 @@ export const getAllProduct = async (req, res) => {
...
-    res.status(500).json({ error: 'Internal server error' });
+    res.status(500).json({ error: "Internal server error" });
```
Comment (Emile-SHUMBUSHO) [inline src/controllers/product.controller.js]: better to user `error:error.message`

File: src/controllers/product.controller.js [hunk 3]
```diff
@@ -105,12 +106,44 @@ export const addProduct = async (req, res) => {
...
+};
+export const getAllForSeller = async (req, res) => {
+  try {
+    const authHeader = req.headers.authorization;
+    if (!authHeader) {
+      return res.status(401).json({ message: "Authorization header missing" });
+    }
+    const token = authHeader.split(" ")[1];
+    const decodedToken = jwt.verify(token, process.env.SECRET_KEY);
+    const sellerId = decodedToken.userId;
```
Comment (Emile-SHUMBUSHO) [inline src/controllers/product.controller.js:119]: this token verification is already created from middleware/verifyToken, it's always a good practice to use it as a middleware, there is no need to rewrite it

**Comment by Emile-SHUMBUSHO** [comment]
Update your readme file, provide documented endpoint screenshot and passing test screenshot if there are available. and remember to resolve conflict from this branch.

**Review by GasanaJr — CHANGES_REQUESTED** [review]
Could you kindly modify the P.R. body, particularly the swagger screenshot, to see the routes in action.

**Review by nshutijonathan — APPROVED** [review]
LGTM

**Review by Emile-SHUMBUSHO — APPROVED** [review]

**Review by GasanaJr — APPROVED** [review]

**Review by Habinezajanvier — APPROVED** [review]
🥇

**Pull request merged**